integration-tests/itest/deploy_test.go

```diff
@@ -0,0 +1,227 @@
+			Expect(streamlets).To(ContainElement(streamlet))
+		}
+		It("should contain a spark process", func() {
+			checkContainsStreamlet("spark-process")
+		})
+
+		It("should contain a flink process", func() {
+			checkContainsStreamlet("flink-process")
+		})
+
+		It("should contain an akka process", func() {
+			checkContainsStreamlet("akka-process")
+		})
+	})
+
+	Context("Running streamlets from the sample app should produce counter data", func() {
+		checkLogsForOutput := func(streamlet string, output string) {
+			pod, err := cli.GetOneOfThePodsForStreamlet(swissKnifeApp, streamlet)
+			Expect(err).NotTo(HaveOccurred())
+			_, err = kctl.PollUntilLogsContains(pod, swissKnifeApp.Name, output)
+			Expect(err).NotTo(HaveOccurred())
+		}
+		It("should produce a counter in the raw output log", func(done Done) {
+			checkLogsForOutput("raw-egress", "count:")
+			close(done)
+		}, LongTimeout)
+
+		It("should produce a counter in the akka output log", func(done Done) {
+			checkLogsForOutput("akka-egress", "count:")
+			close(done)
+		}, LongTimeout)
+
+		It("should produce a counter in the spark output log", func(done Done) {
+			checkLogsForOutput("spark-egress", "count:")
+			close(done)
+		}, LongTimeout)
+
+		It("should produce a counter in the flink output log", func(done Done) {
+			checkLogsForOutput("flink-egress", "count:")
+			close(done)
+		}, LongTimeout)
+	})
+
+	Context("A deployed application can be undeployed", func() {
+		It("should undeploy the test app", func(done Done) {
+			err := cli.Undeploy(swissKnifeApp)
+			Expect(err).NotTo(HaveOccurred())
+			err = PollUntilAppPresenceIs(swissKnifeApp, false)
+			Expect(err).NotTo(HaveOccurred())
+			ensureNoPods(swissKnifeApp)
+			close(done)
+		}, LongTimeout)
+	})
+})
+
+// PollUntilAppPresenceIs polls the list API for the presence of the app.
+// The end condition depends on the `expected` flag
+// If `expected` is true, this method will poll until the app is found (present)
+// If `expected` is false, this method will poll until the app is not found (absent)
+func PollUntilAppPresenceIs(app cli.App, expected bool) error {
+	for {
+		apps, err := cli.ListApps()
+		if err != nil {
+			return err
+		}
+		found := false
+		for _, entry := range apps {
+			if entry.Name == app.Name {
+				found = true
+				break
+			}
+		}
+		if found == expected {
+			return nil
+		}
+		time.Sleep(time.Second)
+	}
+}
+
+func listAppNames(apps []cli.AppEntry) []string {
+	list := make([]string, len(apps))
+	for _, entry := range apps {
+		list = append(list, entry.Name)
+	}
+	return list
+}
+
+func ensureAppNotDeployed(app cli.App) error {
+	apps, err := cli.ListApps()
+	if err != nil {
+		return err
+	}
+	found := false
+	for _, entry := range apps {
+		if entry.Name == app.Name {
+			found = true
+			break
+		}
+	}
+	if found {
+		if err := cli.Undeploy(app); err != nil {
+			return err
+		}
+		return ensureNoPods(app)
+	}
+	return nil
+}
+
+func ensureNoPods(app cli.App) error {
```

Comment:
This seems like a generally useful function to be reused on other tests. I suggest we extract it to `kctl.go`